apps/wallet/src/shared/deepBook/context.tsx

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { useActiveAccount } from '_app/hooks/useActiveAccount';
+import { useGetOwnedObjects } from '@mysten/core';
+import { useSuiClient } from '@mysten/dapp-kit';
+import { DeepBookClient } from '@mysten/deepbook';
+import { createContext, useContext, useMemo, type ReactNode } from 'react';
+
+type DeepBookContextProps = {
+	client: DeepBookClient;
+	accountCapId: string;
+};
+
+const DeepBookContext = createContext<DeepBookContextProps | null>(null);
+
+interface DeepBookContextProviderProps {
+	children: ReactNode;
+}
+
+export function useDeepBookContext() {
+	const context = useContext(DeepBookContext);
+	if (!context) {
+		throw new Error('useDeepBookContext must be used within a DeepBookContextProvider');
+	}
+	return context;
+}
+
+export function DeepBookContextProvider({ children }: DeepBookContextProviderProps) {
+	const suiClient = useSuiClient();
+	const activeAccount = useActiveAccount();
+	const activeAccountAddress = activeAccount?.address;
+
+	const { data } = useGetOwnedObjects(
+		activeAccountAddress,
+		{
+			MatchAll: [{ StructType: '0xdee9::custodian_v2::AccountCap' }],
+		},
+		1,
+	);
+
+	const accountCapId = useMemo(() => {
```

Comment:
Why do we need to memo this? We just access an object field in here right?

Reply:
Updated, it was my misunderstanding of the previous comment